ADBDEV-3999: ORCA generates incorrect plan with exists clause for partitioned table

RekGRpth opened the pull request:
ORCA generates incorrect plan with exists clause for partitioned table

For subqueries with EXISTS, the [first column is always selected](https://github.com/arenadata/gpdb/blob/bb5a6b85b52161d40aa081dc2c5ec9561eabdb65/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp#L1302-L1304) by ORCA.
But if this column does not participate anywhere else in the query and is neither a system nor a distribution key, then it is not marked as used.
Therefore, [when translating to DXL, such a column is skipped](https://github.com/arenadata/gpdb/blob/6ed03a1d78ef8ab6560dc9a5b40a47cd75faaa75/src/backend/gporca/libgpopt/src/translate/CTranslatorExprToDXL.cpp#L7573-L7585), and in a debug build, such a plan is generally not allowed!
On a release build, when a column is skipped, that column is not added to [table_descr](https://github.com/arenadata/gpdb/blob/6ed03a1d78ef8ab6560dc9a5b40a47cd75faaa75/src/backend/gporca/libgpopt/src/translate/CTranslatorExprToDXL.cpp#L7605C3-L7605C14).
And therefore, its number is not inserted into the [m_colid_to_attno_map](https://github.com/arenadata/gpdb/blob/6ed03a1d78ef8ab6560dc9a5b40a47cd75faaa75/src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp#L4699) hashmap of the correspondence of the column number to the attribute number.
And now the attribute number [is not found](https://github.com/arenadata/gpdb/blob/6ed03a1d78ef8ab6560dc9a5b40a47cd75faaa75/src/backend/gpopt/translate/CMappingColIdVarPlStmt.cpp#L140) by the column number.
And so a variable with an [unknown](https://github.com/arenadata/gpdb/blob/6ed03a1d78ef8ab6560dc9a5b40a47cd75faaa75/src/backend/gpopt/translate/CMappingColIdVarPlStmt.cpp#L222-L226) attribute number is created.
Such a variable with an unknown attribute number is treated as a [whole-row](https://github.com/arenadata/gpdb/blob/6ed03a1d78ef8ab6560dc9a5b40a47cd75faaa75/src/backend/executor/execQual.c#L5391-L5399).
In this case, the oid of this variable does not correspond to the [composite type](https://github.com/arenadata/gpdb/blob/6ed03a1d78ef8ab6560dc9a5b40a47cd75faaa75/src/backend/utils/cache/typcache.c#L754-L758) and an error occurs.
The solution to this problem is to limit the output columns to only those referenced in the query for partitioned tables, as is done for non-partitioned tables.
The Union-On-HJNs test changed because in its plan the unused column B of the partitioned table disappeared.

Reproduction on release build:
```sql
CREATE TABLE offers (
    id int,
    product int,
    date date
) DISTRIBUTED BY (id);

INSERT INTO offers SELECT 0, 0, '2023-01-01';

CREATE TABLE contacts (
    contact text,
    id int,
    date date
) DISTRIBUTED BY (id) PARTITION BY RANGE(date) (START (date '2023-01-01') INCLUSIVE END (date '2023-02-01') EXCLUSIVE EVERY (INTERVAL '1 month'));

INSERT INTO contacts SELECT '0', 0, '2023-01-01';

EXPLAIN (ANALYZE on, COSTS off, VERBOSE on)
SELECT id FROM offers WHERE EXISTS (
    SELECT id FROM contacts WHERE product = 0 OR contacts.id = offers.id
);

ERROR:  type text is not composite  (seg1 slice2 172.19.0.3:6435 pid=55051)
```
On a debug build, ORCA can't build a plan.
```sql
CTranslatorExprToDXL.cpp:7580: Failed assertion: !reqd_prop_plan->PcrsRequired()->FMember(colref)
LOG:  Pivotal Optimizer (GPORCA) failed to produce plan (unexpected)
```
SIGSEGV, Segmentation fault on release build:
```sql
CREATE TABLE offers (
    id int,
    product int,
    date date
) DISTRIBUTED BY (id);
INSERT INTO offers SELECT i/100, i, '2023-01-01'::date + (i||' min')::interval FROM generate_series(1, 1000) i;

CREATE TABLE contacts (
    contact int,
    id int,
    date date
) DISTRIBUTED BY (id) PARTITION BY RANGE(date) (START (date '2023-01-01') INCLUSIVE END (date '2023-04-01') EXCLUSIVE EVERY (INTERVAL '1 month'));
INSERT INTO contacts SELECT i, i/100, '2023-01-01'::date + (i||' min')::interval FROM generate_series(1, 1000) i;

CREATE INDEX ON contacts USING bitmap(id);
EXPLAIN (COSTS off, VERBOSE on)
SELECT id, EXISTS (
    SELECT id FROM contacts WHERE id = 1
) FROM offers;
```
```
#0  0x00007f3cf1c0f4fb in raise () from /lib64/libpthread.so.0
#1  0x0000000000e43593 in StandardHandlerForSigillSigsegvSigbus_OnMainThread (processName=0x158a193 "Master process", postgres_signal_arg=11) at elog.c:5588
#2  0x0000000000ccc6c1 in CdbProgramErrorHandler (postgres_signal_arg=11) at postgres.c:3587
#3  <signal handler called>
#4  gpos::CDynamicPtrArray<gpdxl::CDXLTranslateContext const, &gpos::CleanupNULL<gpdxl::CDXLTranslateContext const> >::operator[] (this=0x7f43b00, pos=0) at ../../../../src/backend/gporca/libgpos/include/gpos/common/CDynamicPtrArray.h:319
#5  0x00000000012a4fd4 in gpdxl::CMappingColIdVarPlStmt::VarFromDXLNodeScId (this=0x7fffe27247d0, dxlop=0x99ee168) at CMappingColIdVarPlStmt.cpp:151
#6  0x00000000012bcd3b in gpdxl::CTranslatorDXLToScalar::TranslateDXLScalarIdentToScalar (this=0x7f3be70, scalar_id_node=0x7f47908, colid_var=0x7fffe27247d0) at CTranslatorDXLToScalar.cpp:2025
#7  0x00000000012b7f77 in gpdxl::CTranslatorDXLToScalar::TranslateDXLToScalar (this=0x7f3be70, dxlnode=0x7f47908, colid_var=0x7fffe27247d0) at CTranslatorDXLToScalar.cpp:178
#8  0x00000000012f0ebd in gpdxl::CTranslatorDXLToPlStmt::TranslateDXLProjList (this=0x7fffe27251c0, project_list_dxlnode=0x9b5eb68, base_table_context=0x7fffe2724910, child_contexts=0x7f43b00, output_context=0x7fffe2724a50) at CTranslatorDXLToPlStmt.cpp:4820
#9  0x00000000012f1a7d in gpdxl::CTranslatorDXLToPlStmt::TranslateProjListAndFilter (this=0x7fffe27251c0, project_list_dxlnode=0x9b5eb68, filter_dxlnode=0x9b614b8, base_table_context=0x7fffe2724910, child_contexts=0x7f43b00, targetlist_out=0x7f5e318, qual_out=0x7fffe2724908, output_context=0x7fffe2724a50)
    at CTranslatorDXLToPlStmt.cpp:5145
#10 0x00000000012f3352 in gpdxl::CTranslatorDXLToPlStmt::TranslateDXLBitmapTblScan (this=0x7fffe27251c0, bitmapscan_dxlnode=0x9b61b58, output_context=0x7fffe2724a50, ctxt_translation_prev_siblings=0x7f43b00) at CTranslatorDXLToPlStmt.cpp:5782
#11 0x00000000012e4097 in gpdxl::CTranslatorDXLToPlStmt::TranslateDXLOperatorToPlan (this=0x7fffe27251c0, dxlnode=0x9b61b58, output_context=0x7fffe2724a50, ctxt_translation_prev_siblings=0x7f43b00) at CTranslatorDXLToPlStmt.cpp:344
...
```

RekGRpth committed:
ORCA generates incorrect plan with exists clause for partitioned table

For subqueries with EXISTS, the first column is always selected by ORCA.
But if this column does not participate anywhere else in the query and
is neither a system nor a distribution key, then it is not marked as used.
Therefore, when translating to DXL, such a column is skipped, and
in a debug build, such a plan is generally not allowed!
On a release build, when a column is skipped,
that column is not added to table_descr.
And therefore, its number is not inserted into the m_colid_to_attno_map
hashmap of the correspondence of the column number to the attribute number.
And now the attribute number is not found by the column number.
And so a variable with an unknown attribute number is created.
Such a variable with an unknown attribute number is treated as a whole-row.
In this case, the oid of this variable does not
correspond to the composite type and an error occurs.

The solution to the problem is very simple -
always mark the selected column as used.

RekGRpth committed:
Merge branch 'adb-6.x-dev' into ADBDEV-3999

HustonMmmavr commented on src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp:
```diff
@@ -1934,6 +1934,7 @@ CSubqueryHandler::FRemoveExistentialSubquery(
 		// for existential subqueries, any column produced by inner expression
 		// can be used to check for empty answers; we use first column for that
 		CColRef *colref = pexprInner->DeriveOutputColumns()->PcrFirst();
+		colref->MarkAsUsed();
```
Does the same changes are needed [here](https://github.com/arenadata/gpdb/blob/048a7eeefd162760ae9526de2c0ac37bdb86b9dc/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp#L1304)?

RekGRpth replied:
> Does the same changes are needed [here](https://github.com/arenadata/gpdb/blob/048a7eeefd162760ae9526de2c0ac37bdb86b9dc/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp#L1304)?

It seems, yes, I added it, but I could not find a request for verification, because in the test request, it gets here already after the forced usage marking.

InnerLife0 replied:
The patch is overall OK.

Despite it looks very logical, we can't change the code without the understanding of what it changes.
Try to find the query, which use the same block of code. Then change it to reproduce an error. If you can't find appropriate query, try to add WARNING message to this block of code and run regression tests. The example queries will reveal in regression diffs.
If we can't find a query, or if target column is always already marked as used (or etc) it's better to not touch the code, but leave here a comment about why marking the column is not necessary.

In the latest version of patch, I removed CSubqueryHandler.cpp:1938 line and patch still works. So, it's probably questionable where we should place a fix.

RekGRpth replied:
How about another approach?

InnerLife0 replied:
> How about another approach?

Any comment about why you decided to change your mind and why this solution is better?

RekGRpth replied:
I didn't change my mind, I just made the solution more general by marking the attribute as used before putting it in the array.

InnerLife0 replied:
> I didn't change my mind

You changed your mind about the place of the fix. Why the new place is better and why it's ok to change the function which is used in many different places? Tell me your logic.

RekGRpth replied:
> Why the new place is better

I think the fix at this point might cover other cases not yet found.

InnerLife0 replied:
> Despite it looks very logical, we can't change the code without the understanding of what it changes.

Then we back to the quoted statement.

RekGRpth replied:
We hit CSubqueryHandler.cpp:1936 from CXformSubqueryUnnest:185 and then hit CSubqueryHandler.cpp:1304 from CXformSubqueryUnnest:186 just because of the last fEnforceCorrelatedApply argument.
Therefore, it is enough to add a usage mark after CSubqueryHandler.cpp:1936 and then the used attribute will always come to CSubqueryHandler.cpp:1304.

RekGRpth replied:
So I can only keep the first commit in the pull request and delete the rest.

InnerLife0 replied:
Did you check that we always meet CSubqueryHandler.cpp:1304 after CSubqueryHandler.cpp:1936 for the all queries we know?
If yes, then OK. Let's revert latest changes.

RekGRpth replied:
> we always meet CSubqueryHandler.cpp:1304 after CSubqueryHandler.cpp:1936 for the all queries we know?

No, for example:

[CSubqueryHandler::FCreateCorrelatedApplyForQuantifiedSubquery](https://github.com/arenadata/gpdb/blob/3756bbba81d9e01ee5a8b4d2238734c47fa0c43e/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp#L1227-L1229) -> [CSubqueryHandler::FRemoveExistsSubquery](https://github.com/arenadata/gpdb/blob/3756bbba81d9e01ee5a8b4d2238734c47fa0c43e/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp#L1981-L1986) -> [CSubqueryHandler::FCreateCorrelatedApplyForExistentialSubquery](https://github.com/arenadata/gpdb/blob/3756bbba81d9e01ee5a8b4d2238734c47fa0c43e/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp#L1304)
while
[CXformUtils::SubqueryAnyToAgg](https://github.com/arenadata/gpdb/blob/1c2c49c78e9de67ec2abe5ffe12759a4ec550f23/src/backend/gporca/libgpopt/src/xforms/CXformUtils.cpp#L842-L845) -> [CSubqueryHandler::FRemoveExistsSubquery](https://github.com/arenadata/gpdb/blob/3756bbba81d9e01ee5a8b4d2238734c47fa0c43e/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp#L1988-L1990) -> [CSubqueryHandler::FRemoveExistentialSubquery](https://github.com/arenadata/gpdb/blob/3756bbba81d9e01ee5a8b4d2238734c47fa0c43e/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp#L1936)

InnerLife0 replied:
What query can hit only CSubqueryHandler.cpp:1304? Did you test such query trying to modify and break it?

RekGRpth replied:
with diff
```diff
diff --git a/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp b/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp
index 2ce541a472..bd454265c0 100644
--- a/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp
+++ b/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp
@@ -58,6 +58,8 @@
 
 using namespace gpopt;
 
+static int hit = 0;
+
 #ifdef GPOS_DEBUG
 //---------------------------------------------------------------------------
 //	@function:
@@ -1302,6 +1304,8 @@ CSubqueryHandler::FCreateCorrelatedApplyForExistentialSubquery(
 	// for existential subqueries, any column produced by inner expression
 	// can be used to check for empty answers; we use first column for that
 	CColRef *colref = pexprInner->DeriveOutputColumns()->PcrFirst();
+	hit--;
+	GPOS_RTL_ASSERT(0 == hit);
 
 	pexprInner->AddRef();
 	if (EsqctxtFilter == esqctxt)
@@ -1934,6 +1938,7 @@ CSubqueryHandler::FRemoveExistentialSubquery(
 		// for existential subqueries, any column produced by inner expression
 		// can be used to check for empty answers; we use first column for that
 		CColRef *colref = pexprInner->DeriveOutputColumns()->PcrFirst();
+		hit++;
 
 		if (COperator::EopScalarSubqueryExists == op_id)
 		{
```
the string CSubqueryHandler.cpp:1304 is reached before the string CSubqueryHandler.cpp:1936 on query from unit test DynamicIndexGet-OuterRefs.mdp

RekGRpth replied:
> What query can hit only CSubqueryHandler.cpp:1304?

```diff
diff --git a/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp b/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp
index 2ce541a472..bd454265c0 100644
--- a/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp
+++ b/src/backend/gporca/libgpopt/src/xforms/CSubqueryHandler.cpp
@@ -58,6 +58,8 @@
 
 using namespace gpopt;
 
+static int hit = 0;
+
 #ifdef GPOS_DEBUG
 //---------------------------------------------------------------------------
 //	@function:
@@ -1302,6 +1304,8 @@ CSubqueryHandler::FCreateCorrelatedApplyForExistentialSubquery(
 	// for existential subqueries, any column produced by inner expression
 	// can be used to check for empty answers; we use first column for that
 	CColRef *colref = pexprInner->DeriveOutputColumns()->PcrFirst();
+	hit--;
+	GPOS_RTL_ASSERT(0 <= hit);
 
 	pexprInner->AddRef();
 	if (EsqctxtFilter == esqctxt)
@@ -1934,6 +1938,7 @@ CSubqueryHandler::FRemoveExistentialSubquery(
 		// for existential subqueries, any column produced by inner expression
 		// can be used to check for empty answers; we use first column for that
 		CColRef *colref = pexprInner->DeriveOutputColumns()->PcrFirst();
+		hit++;
 
 		if (COperator::EopScalarSubqueryExists == op_id)
 		{
```
shows that we can hit only CSubqueryHandler.cpp:1304 on query from unit test DynamicIndexGet-OuterRefs.mdp or PartTbl-AggWithExistentialSubquery.mdp or ExistentialSubquriesInsideScalarExpression.mdp

RekGRpth replied:
> Did you test such query trying to modify and break it?

Yes, I changed the queries from all three tests, got an unused column on CSubqueryHandler.cpp:1304, but the CTranslatorExprToDXL.cpp:7580 assert did not work in all three cases, because reqd_prop_plan did not contain this column.

RekGRpth replied:
Thus, another solution suggests itself: remove unused columns from reqd_prop_plan or do not place them into there.

RekGRpth replied:
I suggest another approach, which gets first used column for exist subquery

RekGRpth committed:
fix

RekGRpth committed:
Merge branch 'adb-6.x-dev' into ADBDEV-3999

BenderArenadata commented:
Allure report

RekGRpth committed:
another approach

RekGRpth committed:
Merge branch 'adb-6.x-dev' into ADBDEV-3999

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/53148

RekGRpth committed:
get first used column for exist subquery

RekGRpth committed:
Merge branch 'adb-6.x-dev' into ADBDEV-3999

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/53651

RekGRpth committed:
fix regress test

RekGRpth committed:
fix

RekGRpth committed:
comment

RekGRpth committed:
split

RekGRpth committed:
comment

RekGRpth committed:
simplify

HustonMmmavr commented on src/backend/gporca/libgpopt/src/base/CColRefSet.cpp:
```diff
@@ -126,6 +126,31 @@ CColRefSet::PcrFirst() const
+		}
+	}
+
+	return PcrFirst();
```
As I understand the current solution  instead of selecting the first column of table (and marking it as used)  returns the first used column of it.
If I'm right, the next question is here: for ex. next case (seems it's impossible) - there are no selected columns in table (no one of columns marked as used), this function would return unmarked column and the problem still here. Is it so?

RekGRpth replied:
> As I understand the current solution instead of selecting the first column of table (and marking it as used) returns the first used column of it.

Exactly

RekGRpth replied:
> there are no selected columns in table (no one of columns marked as used), this function would return unmarked column and the problem still here. Is it so?

May be. In this case, I can mark the first column as used.

RekGRpth replied:
fixed

RekGRpth committed:
mark first as used

RekGRpth committed:
Merge branch 'adb-6.x-dev' into ADBDEV-3999

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/54790

KnightMurloc commented:
have you researched why the problem is not reproduced on an unpartitioned table? it looks like this case should also be affected but it's not.

RekGRpth commented:
> have you researched why the problem is not reproduced on an unpartitioned table? it looks like this case should also be affected but it's not.

The solution is not based on partitioned tables.

KnightMurloc commented:
looks like `DeriveOutputColumns` should return a list of **used** columns, as is done for an unpartitioned table.
https://github.com/arenadata/gpdb/blob/3f979993d4832e4c68a8f0be339903de8de0aa6a/src/backend/gporca/libgpopt/src/operators/CLogicalGet.cpp#L212-L232
whereas in the case of a partitioned table, all columns are returned:
https://github.com/arenadata/gpdb/blob/3f979993d4832e4c68a8f0be339903de8de0aa6a/src/backend/gporca/libgpopt/src/operators/CLogicalDynamicGetBase.cpp#L177-L186

I would suggest bringing the behavior of a partitioned table to the behavior of a non-partitioned one. and thus not marking the column as used in `FCreateCorrelatedApplyForExistentialSubquery` and `FRemoveExistentialSubquery` is valid behavior. what do you think?

RekGRpth commented:
> what do you think?

Applied

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/57831

BenderArenadata commented:
Failed job Orca unittests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/757160

BenderArenadata commented:
Failed job Orca unittests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/757161

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/757164

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/757158

BenderArenadata commented:
Failed job Regression tests with ORCA on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/757159

RekGRpth committed:
Merge branch 'adb-6.x-dev' into ADBDEV-3999

RekGRpth commented:
> bringing the behavior of a partitioned table to the behavior of a non-partitioned one

This approach didn't work because CLogicalDynamicGetBase::DeriveOutputColumns is used not only for exists queries, but also in many other places where unused columns are needed.

RekGRpth commented:
> not marking the column as used in `FCreateCorrelatedApplyForExistentialSubquery` and `FRemoveExistentialSubquery`

I only mark the first column as used if all columns were unused.

RekGRpth commented:
> what do you think?

Since queries with exists always require a used column, the approach I propose covers not only partitioned and non-partitioned tables, but also generally all cases in which the DeriveOutputColumns function will return the unused column first.

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/57838

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/758259

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/758260

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/759684

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/759685

RekGRpth committed:
revork solution

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/57878

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/759682

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/759943

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/759944

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/759683

RekGRpth commented:
I was able to reproduce the query from the Union-On-HJNs test, but was unable to reproduce the same plan.
```sql
create table mpp_r6756 (a int, b int, x int, y int) distributed randomly partition by list (a) (DEFAULT PARTITION o);
create table mpp_s6756 (c int, d int, e int) distributed randomly;
select a, b, count(x), count(y)
from mpp_r6756, mpp_s6756 q, mpp_s6756 w, mpp_s6756 e
where a = q.c and a = w.d and a = e.e
group by 1, 2
union all
select 1, * from (
select a, count(x), count(y)
from mpp_r6756, mpp_s6756 q, mpp_s6756 w, mpp_s6756 e
where a = q.c and a = w.d and a = e.e
group by 1
) a;
```

KnightMurloc commented:
I think we should still mark the received attribute as used, or if we expect that only used attributes will returned then assert it.

RekGRpth commented:
> if we expect that only used attributes will be returned then assert it.

I tried this and 2 tests are follbacks to postgres :(
```sql
explain SELECT * FROM csq_r WHERE exists (SELECT * FROM csq_f(csq_r.a));
explain SELECT * FROM csq_r WHERE not exists (SELECT * FROM csq_f(csq_r.a));
```

KnightMurloc commented:
then let's mark them as used.

RekGRpth committed:
Revert "revork solution"

This reverts commit 0d26ab4d433433f6381ea2935e8e612094c68b3f.

RekGRpth commented:
> then let's mark them as used.

I reverted to my previous solution.

KnightMurloc commented:
Ok. but add PR description and maybe to the code why it works without a patch in the case of non-partitioned tables.

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/57932

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/57933

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/762860

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/762877

RekGRpth commented:
I try more global refactor https://github.com/arenadata/gpdb/pull/642

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/762862

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/762857

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/762878

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/762853

RekGRpth committed:
Merge branch 'adb-6.x-dev' into ADBDEV-3999

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61034

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/925834

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/925835

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/925832

InnerLife0 commented on src/backend/gporca/libgpopt/src/base/CColRefSet.cpp:
```diff
@@ -126,6 +126,38 @@ CColRefSet::PcrFirst() const
+//		CColRefSet::PcrFirstUsed
+//
+//	@doc:
+//		Return first used member
```
...and do something if there are no used members yet.

RekGRpth replied:
added

InnerLife0 commented on src/test/regress/expected/subselect_gp_optimizer.out:
```diff
@@ -509,7 +509,6 @@ explain SELECT * FROM csq_r WHERE exists (SELECT * FROM csq_f(csq_r.a));
          SubPlan 1  (slice1; segments: 3)
            ->  Result  (cost=0.00..0.00 rows=1 width=1)
                  ->  Result  (cost=0.00..0.00 rows=1 width=1)
-                       ->  Result  (cost=0.00..0.00 rows=1 width=1)
```
How the new code affects this node?

RekGRpth replied:
verbose shows that patch removes redundant node:
before
```sql
explain (verbose, costs off)
SELECT * FROM csq_r WHERE exists (SELECT * FROM csq_f(csq_r.a));
                QUERY PLAN                
------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)
   Output: a
   ->  Seq Scan on postgres.csq_r
         Output: a
         Filter: (SubPlan 1)
         SubPlan 1  (slice1; segments: 3)
           ->  Result
                 Output: 
                 ->  Result
                       Output: csq_r.a, 
                       ->  Result
                             Output: true
 Optimizer: Pivotal Optimizer (GPORCA)
 Settings: optimizer=on
(14 rows)

explain (verbose, costs off)
SELECT * FROM csq_r WHERE not exists (SELECT * FROM csq_f(csq_r.a));
                QUERY PLAN                
------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)
   Output: a
   ->  Seq Scan on postgres.csq_r
         Output: a
         Filter: (SubPlan 1)
         SubPlan 1  (slice1; segments: 3)
           ->  Result
                 Output: 
                 ->  Result
                       Output: csq_r.a, 
                       ->  Result
                             Output: true
 Optimizer: Pivotal Optimizer (GPORCA)
 Settings: optimizer=on
(14 rows)
```
after
```sql
explain (verbose, costs off)
SELECT * FROM csq_r WHERE exists (SELECT * FROM csq_f(csq_r.a));
                QUERY PLAN                
------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)
   Output: a
   ->  Seq Scan on postgres.csq_r
         Output: a
         Filter: (SubPlan 1)
         SubPlan 1  (slice1; segments: 3)
           ->  Result
                 Output: csq_r.a
                 ->  Result
                       Output: true
 Optimizer: Pivotal Optimizer (GPORCA)
 Settings: optimizer=on
(12 rows)

explain (verbose, costs off)
SELECT * FROM csq_r WHERE not exists (SELECT * FROM csq_f(csq_r.a));
                QUERY PLAN                
------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)
   Output: a
   ->  Seq Scan on postgres.csq_r
         Output: a
         Filter: (SubPlan 1)
         SubPlan 1  (slice1; segments: 3)
           ->  Result
                 Output: csq_r.a
                 ->  Result
                       Output: true
 Optimizer: Pivotal Optimizer (GPORCA)
 Settings: optimizer=on
(12 rows)
```

RekGRpth replied:
change removed

InnerLife0 commented on src/test/regress/sql/qp_correlated_query.sql:
```diff
@@ -834,6 +834,28 @@ DROP TABLE skip_correlated_t4;
+    date date
+) DISTRIBUTED BY (id) PARTITION BY RANGE(date) (START (date '2023-01-01') INCLUSIVE END (date '2023-02-01') EXCLUSIVE EVERY (INTERVAL '1 month'));
+INSERT INTO contacts SELECT '0', 0, '2023-01-01';
+EXPLAIN (COSTS off, VERBOSE on)
```
This tests works even if second PcrFirst() is not replaced with PcrFirstUsed(). How can I test that replacing of second  PcrFirst() occurrence is correct?

RekGRpth replied:
code reverted

RekGRpth committed:
comment expanded

RekGRpth committed:
remove first change and add assert

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61225

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/935664

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/935665

RekGRpth committed:
revert

RekGRpth committed:
Merge branch 'adb-6.x-dev' into ADBDEV-3999

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61230

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/935763

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/935764

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/935761

RekGRpth committed:
revrite solution to fix partitioned table

RekGRpth commented:
I rewrote the solution and corrected the description of the pull request. The problem only occurs with partitioned tables, so I only fixed them.

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61258

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/937051

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/937052

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/937045

InnerLife0 commented:
You say that only partitioned tables affected by problem and changed a code of a base class. This class is used by both, ordinary Dynamic Get and Bitmap ones. In my own tests I created a query with Bitmap Get, which has Unused column in exist clause too, but this query is not breaks on unpatched version.
So, I have a question: Are queries with Bitmap Get affected too? 
1) If yes, we should include one example in tests file.
2) If no, then why? Probably it's possible to copy a solution from Bitmap Get.

RekGRpth committed:
Merge branch 'adb-6.x-dev' into ADBDEV-3999

RekGRpth committed:
move logic from parent to child class

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61417

BenderArenadata commented:
Failed job ORCA code linter on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/949783

RekGRpth committed:
clang format

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61419

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/949851

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/949852

RekGRpth committed:
add bitmap and index scans to tests

RekGRpth commented:
> Are queries with Bitmap Get affected too?

I added bitmap and index scans to tests

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61571

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/957241

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/957242

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/957235

RekGRpth commented:
> Are queries with Bitmap Get affected too?

Yes, and also with Dynamic Index Get too

InnerLife0 commented:
> > Are queries with Bitmap Get affected too?
> 
> Yes, and also with Dynamic Index Get too

Sorry, but I can't understand. You say that Bitmap Get is affected too, but code was moved from a base class.
More, when I try to run tests on not patched version the second select is successfully planned by orca. What is this test for? Please add some comments about key moments of what are you testing.

RekGRpth committed:
fix bitmap test

RekGRpth committed:
Merge branch 'adb-6.x-dev' into ADBDEV-3999

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61773

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/965737

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/965738

BenderArenadata commented:
Failed job Regression tests with ORCA on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/965732

RekGRpth committed:
move logic from child to parent

RekGRpth committed:
revert

RekGRpth committed:
update tests

RekGRpth committed:
fix tests

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/61792

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/966961

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/966962

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/966955

BenderArenadata commented:
Failed job Regression tests with ORCA on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/966956

BenderArenadata commented:
Failed job Regression tests with ORCA on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/967721

RekGRpth commented:
Modifying the `CLogicalDynamicBitmapTableGet` class does not solve the problem with the `bitmap` index, just as modifying the `CLogicalDynamicIndexGet` class does not solve the problem with the `btree` index, while modifying the `CLogicalDynamicGet` class, as well as the base class `CLogicalDynamicGetBase`, solves problems with both indexes and without indexes at all!

Modifications to index classes contain buggy alternatives, e.g.
```diff
diff --git a/src/backend/gporca/libgpopt/include/gpopt/operators/CLogicalDynamicIndexGet.h b/src/backend/gporca/libgpopt/include/gpopt/operators/CLogicalDynamicIndexGet.h
index e9be53c814..d569754c23 100644
--- a/src/backend/gporca/libgpopt/include/gpopt/operators/CLogicalDynamicIndexGet.h
+++ b/src/backend/gporca/libgpopt/include/gpopt/operators/CLogicalDynamicIndexGet.h
@@ -134,6 +134,10 @@ public:
 	virtual COperator *PopCopyWithRemappedColumns(
 		CMemoryPool *mp, UlongToColRefMap *colref_mapping, BOOL must_exist);
 
+	// derive output columns
+	virtual CColRefSet *DeriveOutputColumns(CMemoryPool *mp,
+											CExpressionHandle &exprhdl);
+
 	//-------------------------------------------------------------------------------------
 	// Required Relational Properties
 	//-------------------------------------------------------------------------------------
diff --git a/src/backend/gporca/libgpopt/src/operators/CLogicalDynamicIndexGet.cpp b/src/backend/gporca/libgpopt/src/operators/CLogicalDynamicIndexGet.cpp
index 2c6b2919f9..2bafee3be6 100644
--- a/src/backend/gporca/libgpopt/src/operators/CLogicalDynamicIndexGet.cpp
+++ b/src/backend/gporca/libgpopt/src/operators/CLogicalDynamicIndexGet.cpp
@@ -180,6 +180,38 @@ CLogicalDynamicIndexGet::PopCopyWithRemappedColumns(
 		ppartcnstrRel);
 }
 
+
+//---------------------------------------------------------------------------
+//	@function:
+//		CLogicalDynamicGet::DeriveOutputColumns
+//
+//	@doc:
+//		Derive output columns
+//
+//---------------------------------------------------------------------------
+CColRefSet *
+CLogicalDynamicIndexGet::DeriveOutputColumns(CMemoryPool *mp,
+											CExpressionHandle &	 // exprhdl
+)
+{
+	CColRefSet *pcrs = GPOS_NEW(mp) CColRefSet(mp);
+	for (ULONG i = 0; i < m_pdrgpcrOutput->Size(); i++)
+	{
+		// We want to limit the output columns to only those which are referenced in the query
+		// We will know the entire list of columns which are referenced in the query only after
+		// translating the entire DXL to an expression. Hence we should not limit the output columns
+		// before we have processed the entire DXL.
+		if ((*m_pdrgpcrOutput)[i]->GetUsage() == CColRef::EUsed ||
+			(*m_pdrgpcrOutput)[i]->GetUsage() == CColRef::EUnknown)
+		{
+			pcrs->Include((*m_pdrgpcrOutput)[i]);
+		}
+	}
+
+	return pcrs;
+}
+
+
 // Checking if index is partial given the table descriptor and mdid of the index
 BOOL
 CLogicalDynamicIndexGet::IsPartialIndex(CTableDescriptor *ptabdesc,
```
```sql
CREATE TABLE offers (
    id int,
    product int,
    date date
) DISTRIBUTED BY (id);
INSERT INTO offers SELECT i/100, i, '2023-01-01'::date + (i||' min')::interval FROM generate_series(1, 1000) i;

CREATE TABLE contacts (
    contact int,
    id int,
    date date
) DISTRIBUTED BY (id) PARTITION BY RANGE(date) (START (date '2023-01-01') INCLUSIVE END (date '2023-04-01') EXCLUSIVE EVERY (INTERVAL '1 month'));
INSERT INTO contacts SELECT i, i/100, '2023-01-01'::date + (i||' min')::interval FROM generate_series(1, 1000) i;

set optimizer_enforce_subplans = on;
set optimizer_enumerate_plans = on;

CREATE INDEX ON contacts USING btree(id);

set optimizer_plan_id = 13;
EXPLAIN (COSTS off, VERBOSE on)
SELECT id FROM offers WHERE EXISTS (
    SELECT id FROM contacts WHERE id = 1
);
                                               QUERY PLAN                                               
--------------------------------------------------------------------------------------------------------
 Result
   Output: offers.id
   Filter: (SubPlan 1)
   ->  Gather Motion 3:1  (slice1; segments: 3)
         Output: offers.id
         ->  Seq Scan on postgres.offers
               Output: offers.id
   SubPlan 1  (slice0)
     ->  Limit
           Output: contacts.*
           ->  Materialize
                 Output: contacts.*
                 ->  Gather Motion 3:1  (slice2; segments: 3)
                       Output: contacts.*
                       ->  Limit
                             Output: contacts.*
                             ->  Result
                                   Output: contacts.*
                                   ->  Sequence
                                         Output: contacts.*, contacts.id
                                         ->  Partition Selector for contacts (dynamic scan id: 1)
                                               Partitions selected: 3 (out of 3)
                                         ->  Dynamic Seq Scan on postgres.contacts (dynamic scan id: 1)
                                               Output: contacts.*, contacts.id
                                               Filter: (contacts.id = 1)
 Optimizer: Pivotal Optimizer (GPORCA)
 Settings: optimizer=on
(27 rows)

set optimizer_plan_id = 14;
EXPLAIN (COSTS off, VERBOSE on)
SELECT id FROM offers WHERE EXISTS (
    SELECT id FROM contacts WHERE id = 1
);
                                            QUERY PLAN                                            
--------------------------------------------------------------------------------------------------
 Result
   Output: offers.id
   Filter: (SubPlan 1)
   ->  Gather Motion 3:1  (slice1; segments: 3)
         Output: offers.id
         ->  Seq Scan on postgres.offers
               Output: offers.id
   SubPlan 1  (slice0)
     ->  Limit
           Output: contacts.*
           ->  Materialize
                 Output: contacts.*
                 ->  Gather Motion 3:1  (slice2; segments: 3)
                       Output: contacts.*
                       ->  Result
                             Output: contacts.*
                             ->  Sequence
                                   Output: contacts.*, contacts.id
                                   ->  Partition Selector for contacts (dynamic scan id: 1)
                                         Partitions selected: 3 (out of 3)
                                   ->  Dynamic Seq Scan on postgres.contacts (dynamic scan id: 1)
                                         Output: contacts.*, contacts.id
                                         Filter: (contacts.id = 1)
 Optimizer: Pivotal Optimizer (GPORCA)
 Settings: optimizer=on
(25 rows)

set optimizer_plan_id = 27;
EXPLAIN (COSTS off, VERBOSE on)
SELECT id FROM offers WHERE EXISTS (
    SELECT id FROM contacts WHERE id = 1
);
                                                     QUERY PLAN                                                     
--------------------------------------------------------------------------------------------------------------------
 Gather Motion 3:1  (slice3; segments: 3)
   Output: offers.id
   ->  Seq Scan on postgres.offers
         Output: offers.id
         Filter: (SubPlan 1)
         SubPlan 1  (slice3; segments: 3)
           ->  Materialize
                 Output: contacts.*
                 ->  Broadcast Motion 1:3  (slice2)
                       Output: contacts.*
                       ->  Limit
                             Output: contacts.*
                             ->  Gather Motion 3:1  (slice1; segments: 3)
                                   Output: contacts.*
                                   ->  Limit
                                         Output: contacts.*
                                         ->  Result
                                               Output: contacts.*
                                               ->  Sequence
                                                     Output: contacts.*, contacts.id
                                                     ->  Partition Selector for contacts (dynamic scan id: 1)
                                                           Partitions selected: 3 (out of 3)
                                                     ->  Dynamic Seq Scan on postgres.contacts (dynamic scan id: 1)
                                                           Output: contacts.*, contacts.id
                                                           Filter: (contacts.id = 1)
 Optimizer: Pivotal Optimizer (GPORCA)
 Settings: optimizer=on
(27 rows)

set optimizer_plan_id = 28;
EXPLAIN (COSTS off, VERBOSE on)
SELECT id FROM offers WHERE EXISTS (
    SELECT id FROM contacts WHERE id = 1
);
                                                  QUERY PLAN                                                  
--------------------------------------------------------------------------------------------------------------
 Gather Motion 3:1  (slice3; segments: 3)
   Output: offers.id
   ->  Seq Scan on postgres.offers
         Output: offers.id
         Filter: (SubPlan 1)
         SubPlan 1  (slice3; segments: 3)
           ->  Materialize
                 Output: contacts.*
                 ->  Broadcast Motion 1:3  (slice2)
                       Output: contacts.*
                       ->  Limit
                             Output: contacts.*
                             ->  Gather Motion 3:1  (slice1; segments: 3)
                                   Output: contacts.*
                                   ->  Result
                                         Output: contacts.*
                                         ->  Sequence
                                               Output: contacts.*, contacts.id
                                               ->  Partition Selector for contacts (dynamic scan id: 1)
                                                     Partitions selected: 3 (out of 3)
                                               ->  Dynamic Seq Scan on postgres.contacts (dynamic scan id: 1)
                                                     Output: contacts.*, contacts.id
                                                     Filter: (contacts.id = 1)
 Optimizer: Pivotal Optimizer (GPORCA)
 Settings: optimizer=on
(25 rows)
```

RekGRpth committed:
Merge branch 'adb-6.x-dev' into ADBDEV-3999

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/62824

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1011576

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1011577

BenderArenadata commented:
Failed job Regression tests with ORCA on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1011571

InnerLife0 commented:
I'm ready to approve, but we don't have a proper commit message for this. Old message is related to old solution and probably bad formatted.

RekGRpth committed:
ORCA generates incorrect plan with exists clause for partitioned table

For subqueries with EXISTS, the first column is always selected by ORCA.
But if this column does not participate anywhere else in the query and is
neither a system nor a distribution key, then it is not marked as used.
Therefore, when translating to DXL, such a column is skipped, and in
a debug build, such a plan is generally not allowed!
On a release build, when a column is skipped, that
column is not added to table_descr.
And therefore, its number is not inserted into the m_colid_to_attno_map
hashmap of the correspondence of the column number to the attribute number.
And now the attribute number is not found by the column number.
And so a variable with an unknown attribute number is created.
Such a variable with an unknown attribute number is treated as a whole-row.
In this case, the oid of this variable does not correspond
to the composite type and an error occurs.

The solution to this problem is to limit the output columns
to only those referenced in the query for partitioned tables,
as is done for non-partitioned tables.
The Union-On-HJNs test changed because in its plan the unused
column B of the partitioned table disappeared.

RekGRpth commented:
> a proper commit message for this

added empty commit

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/63286

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1033248

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1033249

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1033242

RekGRpth committed:
Merge branch 'adb-6.x-dev' into ADBDEV-3999

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/63349

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1036527

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1036528

dnskvlnk commented on src/backend/gporca/data/dxl/minidump/Union-On-HJNs.mdp:
```diff
@@ -509,7 +509,7 @@
       <dxl:GatherMotion InputSegments="0,1" OutputSegments="-1">
         <dxl:Properties>
-          <dxl:Cost StartupCost="0" TotalCost="5172.042308" Rows="2.000000" Width="32"/>
+          <dxl:Cost StartupCost="0" TotalCost="5172.041950" Rows="2.000000" Width="32"/>
```
Should we update costs in mdp files?

RekGRpth replied:
yes, why not? we excluded one unused column and then costs became chipper

dnskvlnk replied:
I mean don't we ignore them?

RekGRpth replied:
I don't think so

dnskvlnk commented:
Please, update the branch it's out of date

RekGRpth committed:
Merge branch 'adb-6.x-dev' into ADBDEV-3999

Commit:
Merge branch 'adb-6.x-dev' into ADBDEV-3999

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/63746

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1051551

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1051552
